src/PlantUmlClassDiagramGenerator.Library/ClassDiagramGenerator.cs

```diff
@@ -59,6 +59,76 @@ public override void VisitClassDeclaration(ClassDeclarationSyntax node)
             VisitTypeDeclaration(node, () => base.VisitClassDeclaration(node));
         }
 
+        public override void VisitRecordDeclaration(RecordDeclarationSyntax node)
+        {
+            // Code copied from: VisitTypeDeclaration
+            
+            if (SkipInnerTypeDeclaration(node)) { return; }
+
+            relationships.AddInnerclassRelationFrom(node);
+            relationships.AddInheritanceFrom(node);
+
+            var modifiers = GetTypeModifiersText(node.Modifiers);
+            var keyword = (node.Modifiers.Any(SyntaxKind.AbstractKeyword) ? "abstract " : "")
+                + node.Keyword.ToString();
+
+            var typeName = TypeNameText.From(node);
+            var name = typeName.Identifier;
+            var typeParam = typeName.TypeArguments;
+            var type = $"{name}{typeParam}";
+
+            types.Add(name);
+
+            // Write records as: "class <<record>>"
+            WriteLine($"class {type} {modifiers}<<record>> {{");
+
+            nestingDepth++;
+
+            foreach(var parameter in node.ParameterList.Parameters)
```

Comment:
`node.ParameterList` will be null for records with no parameters (e.g. when a record only has explicit properties), so maybe a null check would be beneficial here

Reply:
apparently `Microsoft.CodeAnalysis.CSharp` needed updating - version `4.4.0-2.final` supports `record struct`.